### ISSUE TITLE
SuiteRuntimeClient: allow this to work in Jupyter Notebooks

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc#3020]
This is probably just a matter use using asyncio "properly".

See:

* #3015 
* https://github.com/cylc/cylc/pull/3004#discussion_r266685170

**Comment by oliver-sanders** [comment]
Good find @kinow https://github.com/cylc/cylc/pull/3015#issuecomment-475098280,

I'm guessing this issue is only going to affect us with Jupyter Notebooks (as we would call the `async_request` method from Tornado)?

My main concern with `nest_asyncio` would be:

> I can't speak to how robust it is yet, since it seems to have only existed for a few hours

It's now had a whole two PRs merged, no activity in 2019. It's pretty simple so perhaps it's stagnant because it "it ain't broke, don't fix it".

**Comment by kinow** [comment]
>as we would call the async_request method from Tornado

You nailed it! That definitely works! Doing `response = client_1.async_request('ping_suite')` returned success without the `nest_asyncio`.

>It's now had a whole two PRs merged, no activity in 2019. It's pretty simple so perhaps it's stagnant because it "it ain't broke, don't fix it".

True, and even better if we don't need it then :tada:

**Comment by oliver-sanders** [comment]
It would definitely be nice if Cylc worked in a Jypyter notebook though, perhaps we could detect this and use `nest_asyncio` as a wrapper, unless Jupyter beats us to it...

**Comment by kinow** [comment]
>It would definitely be nice if Cylc worked in a Jypyter notebook though, perhaps we could detect this and use nest_asyncio as a wrapper, unless Jupyter beats us to it...

But users can choose now, whether they would like to use `nest_asyncio`, or simply call the `async_request` too. As long as we add somewhere in our docs a note saying they will need this, I believe we are good now :grin: 

![image](https://user-images.githubusercontent.com/304786/54784561-118a0800-4c89-11e9-88d1-a58ba97980c3.png)

**Comment by hjoliver** [comment]
Yes let's be sure to document this.  Can we keep this issue open until that is done?

**Comment by oliver-sanders** [comment]
Sounds like a plan, have renamed issue to match.
